### PR TITLE
Don't count range download in download limit

### PR DESCRIPTION
**Pull request by CarlSchwan**
I tried to still count the first or the last range but this is not really reliable since we send two range `0-` at the beginning (one for the preview and one for the video) and the last range `xxxxxxxxx-` is also called multiple times

### CODE REVIEW

**Comment by szaimen** [comment]
The fix looks good to me in general. I wonder if we should revert https://github.com/nextcloud/server/pull/28227 now that this is fixed?

**Comment by szaimen** [comment]
@nickvergessen I think it would be good if you would also add your concerns on that PR :)

**Comment by nickvergessen** [comment]
Yeah, I have nothing to say to this. This allows to trick out the download limit completely by just always downloading with range `0-`. Even for text files, documents, images, etc. not only streamed items. Maybe it can be limited to that kind of resources?

File: apps/files_sharing/lib/Controller/ShareController.php [hunk 1]
```diff
@@ -686,7 +686,9 @@ public function downloadShare($token, $files = null, $path = '', $downloadStartS
 		}
 
 		$this->emitAccessShareHook($share);
-		$this->emitShareAccessEvent($share, self::SHARE_DOWNLOAD);
+		if (!isset($_SERVER['HTTP_RANGE'])) {
```
Comment (skjnldsv) [inline apps/files_sharing/lib/Controller/ShareController.php:689]: Please document the use case

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: apps/files_sharing/lib/Controller/ShareController.php (hunk 1 above)
Comment (CarlSchwan) [reply apps/files_sharing/lib/Controller/ShareController.php]: Done

**Comment by skjnldsv** [comment]
> Yeah, I have nothing to say to this. This allows to trick out the download limit completely by just always downloading with range `0-`. Even for text files, documents, images, etc. not only streamed items. Maybe it can be limited to that kind of resources?

@PVince81 since we talked about it this morning.
I think it's ok to not bind this to videos only. What about audio? When does it stop? :see_no_evil: 

The download limit by itself is not a perfect science and we made sure to communicate that. Anyone with decent skills can counter this and download the file then distribute it outside. Even with or without range protection. :shrug:

File: apps/files_sharing/lib/Controller/ShareController.php [hunk 2]
```diff
@@ -652,8 +655,10 @@ public function downloadShare($token, $files = null, $path = '', $downloadStartS
 			$originalSharePath = $userFolder->getRelativePath($node->getPath());
 
 			if ($node instanceof \OCP\Files\File) {
+				$node = $share->getNode();
 				// Single file download
-				$this->singleFileDownloaded($share, $share->getNode());
+				$this->singleFileDownloaded($share, $node);
+				$isVideoAutio = str_starts_with($node->getMimeType(), 'video/') || str_starts_with($node->getMimeType(), 'audio/');
```
Comment (skjnldsv) [inline apps/files_sharing/lib/Controller/ShareController.php:661]:
```suggestion
				$isVideoAudio = str_starts_with($node->getMimeType(), 'video/') || str_starts_with($node->getMimeType(), 'audio/');
```

**Review by szaimen — CHANGES_REQUESTED** [review]
I need an answer on this:
https://github.com/nextcloud/server/pull/35324#issuecomment-1323435010

**Comment by skjnldsv** [comment]
> The fix looks good to me in general. I wonder if we should revert #28227 now that this is fixed?

Might be a good idea yes

**Comment by nickvergessen** [comment]
Why that? The activity spam would just come back?
Also with the change in this PR we can't move the activity app over, as it would not create an event anymore when the viewer is used to look at it.

#28227 was only for activities, not for the download limit counter.

**Comment by PVince81** [comment]
if possible, let's see if switching to a header "X-NC-Viewer" would work there and have the download limit app ignore such requests: https://github.com/nextcloud/files_pdfviewer/pull/654#issuecomment-1323403799

**Pull request closed without merging**